include/drivers/serial.h

```diff
@@ -148,6 +148,24 @@ union interrupt_enable_register {
 };
 typedef union interrupt_enable_register ier_t;
 
+union interrupt_identification_register {
+    uint8_t reg;
+    struct __packed {
+        uint8_t no_int_pend : 1, event : 3, rsvd : 1, fifo_64b : 1, fifo_status : 2;
+    };
```

Comment:
The order of bit field members in C is implementation defined and for gcc it's defined by the ABI.
I'm fine with that, but we probably want a block like this?:
```
#if  __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
// order as above
#else 
// reverse order
#endif
```
Not that it matters now, but in case someone wants to add Aarch64 support or the like.

Reply:
Right, this is a very good point and IMO deserves a dedicated issue not to forget about it. We use similar bit-fields (where order would matter) in several other places too.